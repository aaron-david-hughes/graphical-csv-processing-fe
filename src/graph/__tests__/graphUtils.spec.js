import GraphUtils from "../graphUtils";
import Config from "../../config.json";
import deepClone from "lodash.clonedeep";

describe('graphUtils tests', () => {
    let testGetData = function(data, key) {
        return data[key];
    }

    describe('fileToolTip tests', () => {
        let fileNodeData;

        it('should render html with no errors when valid data', () => {
            fileNodeData = {
                operation: 'open_file',
                name: 'name.csv',
                expectedInputs: '0',
                inputCardinality: '0'
            };

            expect(GraphUtils.fileTooltip(setUpGraph(fileNodeData))).toMatchSnapshot();
        });

        it('should render html with cardinality error when valid data with incorrect input edges', () => {
            fileNodeData = {
                operation: 'open_file',
                name: 'name.csv',
                expectedInputs: '0',
                inputCardinality: '1'
            };

            expect(GraphUtils.fileTooltip(setUpGraph(fileNodeData))).toMatchSnapshot();
        });

        it('should render html with data validity error when invalid data with correct input edges', () => {
            fileNodeData = {
                operation: 'open_file',
                name: null,
                expectedInputs: '0',
                inputCardinality: '0'
            };

            expect(GraphUtils.fileTooltip(setUpGraph(fileNodeData))).toMatchSnapshot();
        });

        it('should render html with both validity and cardinality errors when invalid data with incorrect input edges', () => {
            fileNodeData = {
                operation: 'open_file',
                name: null,
                expectedInputs: '0',
                inputCardinality: '1'
            };

            expect(GraphUtils.fileTooltip(setUpGraph(fileNodeData))).toMatchSnapshot();
        });
    });

    describe('processToolTip tests', () => {
        let processingNodeData;

        it('should render html with no errors when valid data', () => {
            processingNodeData = {
                operation: 'join',
                specificOperation: 'join_left',
                onLeft: 'test1',
                onRight: 'test2',
                joinType: 'left',
                expectedInputs: '2',
                inputCardinality: '2'
            };

            expect(GraphUtils.processTooltip(setUpGraph(processingNodeData), Config)).toMatchSnapshot();
        });

        it('should render html with cardinality error when valid data with incorrect input edges', () => {
            processingNodeData = {
                operation: 'join',
                specificOperation: 'join_left',
                onLeft: 'test1',
                onRight: 'test2',
                joinType: 'left',
                expectedInputs: '2',
                inputCardinality: '1'
            };

            expect(GraphUtils.processTooltip(setUpGraph(processingNodeData), Config)).toMatchSnapshot();
        });

        it('should render html with data validity error when invalid data with correct input edges', () => {
            processingNodeData = {
                operation: 'join',
                specificOperation: 'join_left',
                onLeft: 'test1',
                onRight: null,
                joinType: 'left',
                expectedInputs: '2',
                inputCardinality: '2'
            };

            expect(GraphUtils.processTooltip(setUpGraph(processingNodeData), Config)).toMatchSnapshot();
        });

        it('should render html with no errors when data missing but not configured to be required', () => {
            processingNodeData = {
                operation: 'join',
                specificOperation: 'join_left',
                onLeft: 'test1',
                onRight: null,
                joinType: 'left',
                expectedInputs: '2',
                inputCardinality: '2'
            };

            let nodeConfig = deepClone(Config);
            nodeConfig.processing.operations.find(op => op.operation === 'join_left').onRight.required = false;

            expect(GraphUtils.processTooltip(setUpGraph(processingNodeData), nodeConfig)).toMatchSnapshot();
        });

        it('should render html with both validity and cardinality errors when invalid data with incorrect input edges', () => {
            processingNodeData = {
                operation: 'join',
                specificOperation: 'join_left',
                onLeft: 'test1',
                onRight: null,
                joinType: 'left',
                expectedInputs: '2',
                inputCardinality: '3'
            };

            expect(GraphUtils.processTooltip(setUpGraph(processingNodeData), Config)).toMatchSnapshot();
        });
    });

    function setUpGraph(data) {
        return {
            id: 'testId',
            getData: testGetData.bind(null, data)
        };
    }
});