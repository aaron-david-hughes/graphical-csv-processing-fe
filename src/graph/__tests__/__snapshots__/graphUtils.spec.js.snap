// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphUtils tests fileToolTip tests should render html with both validity and cardinality errors when invalid data with incorrect input edges 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: open_file<br>name: null<br>expectedInputs: 0<br><p style='background-color: #FF272A !important; margin: 2px; padding: 5px; border-radius: 8px;'>Invalid number of inputs: 1, must be: 0.</p><br><p style='background-color: #FF272A !important; margin: 2px; padding: 5px; border-radius: 8px;'>Invalid config, double click to edit the configuration.</p>"`;

exports[`graphUtils tests fileToolTip tests should render html with cardinality error when valid data with incorrect input edges 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: open_file<br>name: name.csv<br>expectedInputs: 0<br><p style='background-color: #FF272A !important; margin: 2px; padding: 5px; border-radius: 8px;'>Invalid number of inputs: 1, must be: 0.</p>"`;

exports[`graphUtils tests fileToolTip tests should render html with data validity error when invalid data with correct input edges 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: open_file<br>name: null<br>expectedInputs: 0<br><p style='background-color: #FF272A !important; margin: 2px; padding: 5px; border-radius: 8px;'>Invalid config, double click to edit the configuration.</p>"`;

exports[`graphUtils tests fileToolTip tests should render html with no errors when valid data 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: open_file<br>name: name.csv<br>expectedInputs: 0"`;

exports[`graphUtils tests processToolTip tests should render html with both validity and cardinality errors when invalid data with incorrect input edges 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: join<br>specificOperation: join_left<br>onLeft: test1<br>onRight: null<br>joinType: left<br>expectedInputs: 2<br><p style='background-color: #FF272A !important; margin: 2px; padding: 5px; border-radius: 8px;'>Invalid config, double click to edit the configuration.</p><br><p style='background-color: #FF272A !important; margin: 2px; padding: 5px; border-radius: 8px;'>Invalid number of inputs: 3, must be: 2.</p>"`;

exports[`graphUtils tests processToolTip tests should render html with cardinality error when valid data with incorrect input edges 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: join<br>specificOperation: join_left<br>onLeft: test1<br>onRight: test2<br>joinType: left<br>expectedInputs: 2<br><p style='background-color: #FF272A !important; margin: 2px; padding: 5px; border-radius: 8px;'>Invalid number of inputs: 1, must be: 2.</p>"`;

exports[`graphUtils tests processToolTip tests should render html with data validity error when invalid data with correct input edges 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: join<br>specificOperation: join_left<br>onLeft: test1<br>onRight: null<br>joinType: left<br>expectedInputs: 2<br><p style='background-color: #FF272A !important; margin: 2px; padding: 5px; border-radius: 8px;'>Invalid config, double click to edit the configuration.</p>"`;

exports[`graphUtils tests processToolTip tests should render html with no errors when data missing but not configured to be required 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: join<br>specificOperation: join_left<br>onLeft: test1<br>onRight: null<br>joinType: left<br>expectedInputs: 2"`;

exports[`graphUtils tests processToolTip tests should render html with no errors when valid data 1`] = `"<span style='font-weight:bold'>testId</span><br><br>operation: join<br>specificOperation: join_left<br>onLeft: test1<br>onRight: test2<br>joinType: left<br>expectedInputs: 2"`;
